### PR TITLE
ET-STM32 Stamp: An STM32F103RET6 target for NuttX

### PR DESCRIPTION
**Pull request by ramangopalan**
Dear list, greetings!

## Summary

This PR brings NuttX to the ET-STM32 Stamp module [1] from Futurlec. I
also utilized this opportunity to get familiar with the NuttX
internals and parts of the ARM codebase. Many thanks for NuttX!

I went through the list of already supported STM32F103 platforms. I
picked Maple as a starting point. I made an et-stm32-stamp directory
and added an initial NSH configuration. I touched boards/Kconfig for
menuconfig.

The Stamp is a very compact PCB with just pinouts. It doesn't have any
on-board peripherals; Not even a GPIO_LED OR a button I can detect
with an interrupt.

The nsh configuration is very minimal for now. It doesn't initialize/utilize
any other MCU peripheral of the on-board STM32F103RET6. I will slowly
start adding newer configurations by reading stm32f103-mimimum/.

## Impact

This PR doesn't conflict with any other board/sub-system. It is
independent.

## Testing

  1) I have tested et-stm32-stamp/nsh using WSL with my Stamp
  module. I have documented the details in the board README file. Can
  you please let me know if the code is OK and if there's anything
  else I can do in this regard?

  2) This PR should pass the CI tests.

Many thanks.

References:
[1]: https://www.futurlec.com/ET-STM32_Stamp.shtml

### CODE REVIEW

File: boards/arm/stm32/et-stm32-stamp/include/board.h [hunk 1]
```diff
@@ -0,0 +1,167 @@
+/****************************************************************************
```
Comment (davids5) [inline boards/arm/stm32/et-stm32-stamp/include/board.h:1]: New submission should use the Apache licence.

File: boards/arm/stm32/et-stm32-stamp/include/board.h [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+ * boards/arm/stm32/et-stm32-stamp/include/board.h
+ * include/arch/board/board.h
```
Comment (davids5) [inline boards/arm/stm32/et-stm32-stamp/include/board.h:3]: This is useful when learning NuttX. I like it, but not to the coding standard.

**Review by davids5 — CHANGES_REQUESTED** [review]
@ramangopalan - looks good except for as noted. On the board files. Greg and signed a SGA, and I am not sure of the status of the other author. But the files are duplicates of others int tree that only to Greg. If you replace with of those, the change to Apache license will a either already be done or OK to do.

**Review by davids5 — APPROVED** [review]
@ramangopalan - Thank you!

**Comment by ramangopalan** [comment]
Many thanks for the suggestions David!

**Pull request merged**